examples/analytics/airflow-on-eks/main.tf

```diff
@@ -0,0 +1,542 @@
+#---------------------------------------------------------------
+resource "random_password" "postgress" {
+  length           = 16
+  special          = true
+  override_special = "![]{}"
+}
+#tfsec:ignore:aws-ssm-secret-use-customer-key
+resource "aws_secretsmanager_secret" "postgres" {
+  name                    = "postgres"
+  recovery_window_in_days = 0 # Set to zero for this example to force delete during Terraform destroy
+}
+
+resource "aws_secretsmanager_secret_version" "postgres" {
+  secret_id     = aws_secretsmanager_secret.postgres.id
+  secret_string = random_password.postgress.result
+}
+
+data "aws_secretsmanager_secret_version" "postgres" {
+  secret_id = aws_secretsmanager_secret.postgres.id
+
+  depends_on = [aws_secretsmanager_secret_version.postgres]
+}
+
+#---------------------------------------------------------------
+# Apache Airflow Webserver Secret
+#---------------------------------------------------------------
+resource "random_password" "airflow_webserver" {
+  length           = 16
+  special          = true
+  override_special = "![]{}"
+}
+#tfsec:ignore:aws-ssm-secret-use-customer-key
+resource "aws_secretsmanager_secret" "airflow_webserver" {
+  name                    = "airflow_webserver"
+  recovery_window_in_days = 0 # Set to zero for this example to force delete during Terraform destroy
+}
+
+resource "aws_secretsmanager_secret_version" "airflow_webserver" {
+  secret_id     = aws_secretsmanager_secret.airflow_webserver.id
+  secret_string = random_password.airflow_webserver.result
+}
+
+data "aws_secretsmanager_secret_version" "airflow_webserver" {
+  secret_id = aws_secretsmanager_secret.airflow_webserver.id
+
+  depends_on = [aws_secretsmanager_secret_version.airflow_webserver]
+}
+
+#---------------------------------------------------------------
+# Webserver Secret Key
+#---------------------------------------------------------------
+resource "kubectl_manifest" "airflow_webserver" {
+  sensitive_fields = [
+    "data.webserver-secret-key"
+  ]
+
+  yaml_body = <<YAML
+apiVersion: v1
+kind: Secret
+metadata:
+   name: ${local.airflow_webserver_secret_name}
+   namespace: ${module.airflow_irsa.namespace}
+type: Opaque
+data:
+  webserver-secret-key: ${base64encode(data.aws_secretsmanager_secret_version.airflow_webserver.secret_string)}
+YAML
+
+  depends_on = [module.eks_blueprints.eks_cluster_id]
+}
+
+#---------------------------------------------------------------
+# Managing DAG files with GitSync - EFS Storage Class
+#---------------------------------------------------------------
+resource "kubectl_manifest" "efs_sc" {
+  yaml_body = <<YAML
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: efs-sc
+provisioner: efs.csi.aws.com
+parameters:
+  provisioningMode: efs-ap
+  fileSystemId: ${aws_efs_file_system.efs.id}
+  directoryPerms: "700"
+  gidRangeStart: "1000"
+  gidRangeEnd: "2000"
+YAML
+
+  depends_on = [module.eks_blueprints.eks_cluster_id]
```

Comment:
we should be able to remove the dependency on the cluster since the `kubectl` provider will wait for it to be ready to receive requests (it polls), but do we need to ensure the EFS CSI driver is installed first? i.e. - do we need to do depend on the addons instead of the cluster itself?